photon-ml/src/main/scala/com/linkedin/photon/ml/function/SmoothedHingeLossFunction.scala

```diff
@@ -15,22 +15,42 @@
 package com.linkedin.photon.ml.function
 
 import breeze.linalg
+
+import com.linkedin.photon.ml.constants.MathConst
 import com.linkedin.photon.ml.data.LabeledPoint
 
 /**
  * Implement Rennie's smoothed hinge loss function (http://qwone.com/~jason/writing/smoothHinge.pdf) as an
  * optimizer-friendly approximation for linear SVMs
  */
 class SmoothedHingeLossFunction extends DiffFunction[LabeledPoint] {
-
   override protected[ml] def calculateAt(
       datum: LabeledPoint,
       coefficients: linalg.Vector[Double],
       cumGradient: linalg.Vector[Double]): Double = {
 
-    val actualLabel = if (datum.label < 0.5) -1 else 1
     val margin = datum.computeMargin(coefficients)
-    val z = actualLabel * margin
+    val (loss, deriv) = SmoothedHingeLossFunction.lossAndDerivative(margin, datum.label)
+
+    // Eq. 5, page 2 (derivative multiplied by label in lossAndDerivative method)
+    breeze.linalg.axpy(datum.weight * deriv, datum.features, cumGradient)
+    datum.weight * loss
+  }
+}
+
+object SmoothedHingeLossFunction {
```

Comment:
Yes, these changes just move the calculation out to an object so that it is available for the `SmoothedHingeLossEvaluator`, they shouldn't modify the behaviour. Even so, the behaviour is tested by the `ObjectiveFunctionTest` and `ObjectiveFunctionIntegTest` tests.